README.markdown

```diff
@@ -228,6 +229,34 @@ rocommunity shibboleth 192.168.0.0/16
 rocommunity shibboleth 1.2.3.4/32
 ```
 
+### Proxy Configuration
+
+A snmp proxy can be configured to target a other snmp agent depending on the snmp context.
+
+```puppet
+class { 'snmp':
+  proxy   => [ { hostname => 'dv0',
+                 context => 'ctx_host1',
+                 version =>  '3',
+                 extra_opt => '-t 30',
+                 subtree => '.1.3',
+                 users =>{ 'v3user_md5d' => {
+                                              authtype => 'MD5',
+                                              authpass => 'authpass',
+                                              privtype => 'DES',
+                                              privpass => 'privpass',
+                                            }
+                         }
+               },
```

Comment:
This seems terribly complex and enforces the use of SNMPv3 context when it may not be desired.  Why not just an array of strings like `com2sec` or `dlmod`?  I am worried that there is no input validation and that will lead to errors.  Otherwise, we are going to end up with users messing up that data structure and wondering how to make it work.